### PR TITLE
Fix unaligned read and write ASAN error in stub TLS

### PR DESCRIPTION
**Pull request by ThadHouse**
The pointer punning was unaligned, and ASAN warns about this.

This might be whats seen in #537 but I'm not sure. That issue specifically doesn't reproduce for me.

### CODE REVIEW

File: src/platform/tls_stub.c [hunk 1]
```diff
@@ -1900,7 +1902,9 @@ QuicDecrypt(
     UNREFERENCED_PARAMETER(AuthDataLength);
     UNREFERENCED_PARAMETER(AuthData);
     uint16_t PlainTextLength = BufferLength - QUIC_ENCRYPTION_OVERHEAD;
-    if (*(uint64_t*)(Buffer + PlainTextLength) != Key->Secret) {
+    uint64_t ReadVal = 0;
+    QuicCopyMemory(&ReadVal, Buffer + PlainTextLength, sizeof(ReadVal));
+    if (ReadVal != Key->Secret) {
```
Comment (nibanks) [inline src/platform/tls_stub.c:1907]: Use a mem compare instead?

File: src/platform/tls_stub.c [hunk 2]
```diff
@@ -1877,8 +1877,10 @@ QuicEncrypt(
     UNREFERENCED_PARAMETER(AuthDataLength);
     UNREFERENCED_PARAMETER(AuthData);
     uint16_t PlainTextLength = BufferLength - QUIC_ENCRYPTION_OVERHEAD;
-    ((uint64_t*)(Buffer + PlainTextLength))[0] = Key->Secret;
-    ((uint64_t*)(Buffer + PlainTextLength))[1] = 0;
+    uint64_t WriteVal = Key->Secret;
+    QuicCopyMemory(Buffer + PlainTextLength, &WriteVal, sizeof(WriteVal));
```
Comment (nibanks) [inline src/platform/tls_stub.c:1881]: You shouldn't need a temporary variable.

File: src/platform/tls_stub.c [hunk 3]
```diff
@@ -1877,8 +1877,10 @@ QuicEncrypt(
...
+    WriteVal = 0;
+    QuicCopyMemory(Buffer + PlainTextLength + sizeof(WriteVal), &WriteVal, sizeof(WriteVal));
```
Comment (nibanks) [inline src/platform/tls_stub.c:1883]: Just use a mem set instead? Or QuicZeroMemory?

File: src/platform/tls_stub.c (hunk 2 above)
Comment (ThadHouse) [reply src/platform/tls_stub.c:1881]: I needed the temp for the 2nd value. But if I use zero, I can get rid of it.

**Comment by nibanks** [comment]
@ThadHouse we've consistently been seeing some failures with this now:

image

I ran it three times and can't get a clean run. Can you take a look?

**Pull request merged**